Updated default and hover state styling - RSC-679

entroform opened the pull request:
https://issues.sonatype.org/browse/RSC-679

In addition to NxCheckbox and NxRadio changes, I updated `--nx-color-form-element-border` CSS variable with `--nx-swatch-grey-50` to match Kaleido. This might will change the border color for other form elements as side effect.

rpokorny commented:
> This might will change the border color for other form elements as side effect.

You should determine which ones change and whether those changes match kaleido.  One case I see is that this PR changes the hover border on `NxColorPicker` in a way that does not match the spec for that component.  That should be addressed here.

entroform commented:
> > This might will change the border color for other form elements as side effect.
> 
> You should determine which ones change and whether those changes match kaleido. One case I see is that this PR changes the hover border on `NxColorPicker` in a way that does not match the spec for that component. That should be addressed here.

Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by `--nx-color-form-element-border`?

rpokorny commented:
> Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by --nx-color-form-element-border?

The original purpose of the ticket is what it is (and shouldn't be changed at this point IMO) but if other "bugs" (mismatches with kaleido) get fixed in the process of fixing this one it wouldn't hurt to mention them in a comment on the JIRA ticket

glenhunter commented:
I think that `NxRadio` needs the same border adjustment(s)

entroform commented:
> > Thanks, going to look into it and address it. Should I update the ticket to reflect changes affected by --nx-color-form-element-border?
> 
> The original purpose of the ticket is what it is (and shouldn't be changed at this point IMO) but if other "bugs" (mismatches with kaleido) get fixed in the process of fixing this one it wouldn't hurt to mention them in a comment on the JIRA ticket

Fixed other mismatches affected by  `--nx-color-form-element-border` here :)
6b7d61614e64372ad4c8eab6bb09f948817ed8b3

I also noticed that the active state for radios and checkboxes also do no align; I updated it in the above commit.
Let me know if you find any issues with the implementation 👍

glenhunter commented on lib/src/components/NxCheckbox/NxCheckbox.scss:
```diff
@@ -32,11 +32,22 @@ $-width: 16px;
     background-color:var(--nx-color-interactive-background-disabled);
     border-color: var(--nx-color-interactive-border-disabled);
-
+    
```
```suggestion

```

glenhunter commented on lib/src/components/NxCheckbox/NxCheckbox.tsx:
```diff
@@ -56,7 +56,7 @@ const NxCheckbox = forwardRef<HTMLLabelElement, Props>(
               * in order to provide a consistent vertical-align baseline
               */}
-            { isChecked ? <FontAwesomeIcon icon={faCheck} /> : '\u00A0' }
+            <FontAwesomeIcon icon={faCheck} />
```
Two things: is this in scope? if so do we still need the comment above?

entroform replied:
> Two things: is this in scope? if so do we still need the comment above?

Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.

glenhunter replied:
> > Two things: is this in scope? if so do we still need the comment above?
> 
> Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.

Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state

entroform replied:
> > > Two things: is this in scope? if so do we still need the comment above?
> > 
> > 
> > Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.
> 
> Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state

Yeah, I agree that it should only appear in the `:checked` state. I was confused because in the Kaleido the `:active` state is the same as the `:checked` state. I started a conversation about it in the #react-components channel and it looks like @alexalyons126 will be updating it

entroform replied:
> > > > Two things: is this in scope? if so do we still need the comment above?
> > > 
> > > 
> > > Hi Glen, I believe so. This change is needed because the icon needs to be visible in the `active` state as well as the`checked` state according to Kaleido. Nice catch, I forgot to remove the comment.
> > 
> > 
> > Well technically the `:active` state of a checkbox (does a checkbox even have an `active` state?) could be either of the state of checking it or unchecking it. The checkmark icon should only appear when the checkbox is in the `:checked` state
> 
> Yeah, I agree that it should only appear in the `:checked` state. I was confused because in the Kaleido the `:active` state is the same as the `:checked` state. I started a conversation about it in the #react-components channel and it looks like @alexalyons126 will be updating it

After offline discussions, the decision was to revert back to how it was before. 44c1e9043dad5474c028b5d79b166e6b7e7baaef

entroform commented:
I started a conversation about the active state on checkboxes and radios in the #react-components channel.
It looks like @alexalyons126 will be updating it in Kaleido and once that's done I'll update this PR 👍.

rpokorny commented on lib/src/base-styles/_nx-radio-checkbox.scss:
```diff
@@ -48,7 +48,7 @@
     &:hover {
       .nx-radio-checkbox__control {
-        border-color: var(--nx-color-form-element-border);
+        border-color: var(--nx-swatch-grey-05);
         filter: none;
         stroke: var(--nx-swatch-grey-05);
```
I don't think border and stroke are actually necessary here at all; they just repeat what's already present via the non-focus `:hover` selector

entroform replied:
> I don't think border and stroke are actually necessary here at all; they just repeat what's already present via the non-focus `:hover` selector

Thanks! Removed here:
5c5ea6780bb688265db64fa897b7852cd8c5935e

rpokorny commented on lib/src/components/NxCheckbox/NxCheckbox.scss:
```diff
@@ -36,7 +36,6 @@ $-width: 16px;
   + .nx-checkbox__box {
     background-color:var(--nx-color-interactive-background-disabled);
     border-color: var(--nx-color-interactive-border-disabled);
-
```
Extraneous and undesirable change

entroform replied:
> Extraneous and undesirable change

Ops, totally missed this. Fixed:
5c5ea6780bb688265db64fa897b7852cd8c5935e

rpokorny commented on lib/src/components/NxCodeSnippet/NxCodeSnippet.scss:
```diff
@@ -44,7 +44,7 @@
 
     .nx-text-input__box {
-      border-color: var(--nx-color-form-element-border);
+      border-color: var(--nx-swatch-grey-05);
```
@alexalyons126 is this actually what we want? It matches kaleido but effectively what this means is that the text field in an `NxCodeSnippet` would always look the way a _hovered_ text input is supposed to.

entroform replied:
> > @alexalyons126 is this actually what we want? It matches kaleido but effectively what this means is that the text field in an `NxCodeSnippet` would always look the way a _hovered_ text input is supposed to.
> 
> I reached out to @alexalyons126 on Slack to confirm.

Confirmed with @alexalyons126 that this should match other form fields.
Reverted here:
4c97190c807354f3883f3f2691fe6a38a1f3c2e7
